src/compute-client/src/metrics.rs

```diff
@@ -430,8 +454,71 @@ impl StatsCollector<ProtoComputeCommand, ProtoComputeResponse> for ReplicaMetric
 pub(crate) struct ReplicaCollectionMetrics {
     /// Gauge tracking dataflow hydration time.
     pub initial_output_duration_seconds: Gauge,
-    /// Histogram tracking dataflow wallclock lag.
-    pub wallclock_lag_seconds: Histogram,
+    /// Gauge tracking maximum dataflow wallclock lag.
+    wallclock_lag_max_seconds: Gauge,
+    /// Gauge tracking minimum dataflow wallclock lag.
+    wallclock_lag_min_seconds: Gauge,
+    /// Gauge tracking average dataflow wallclock lag.
+    wallclock_lag_avg_seconds: Gauge,
+
+    /// State for aggregating wallclock lag.
+    wallclock_lag: WallclockLag,
+}
+
+impl ReplicaCollectionMetrics {
+    pub fn observe_wallclock_lag(&mut self, lag: Duration) {
+        const REFRESH_INTERVAL: Duration = Duration::from_secs(60);
+
+        self.wallclock_lag.add(lag);
+
+        if self.wallclock_lag.last_reset.elapsed() >= REFRESH_INTERVAL {
```

Comment:
I'm having a little trouble wrapping my head around this. How will we aggregate `lag_avg_seconds` across environments and dataflows? Taking the average of averages is sketchy.

Are you sure we don't want to just expose the sum and count directly as individual metrics, and then compute the average downstream in PromQL? That way we can aggregate however we like and still compute the average second-by-second freshness.

Reply:
Yes. We can solve this either be emitting sum and count or average and count.

Reply:
I guess sum and count is preferable: No precision loss and it guides you away from doing a non-weighted average of averages.